xarray/plot/plot.py

```diff
@@ -646,6 +648,28 @@ def newplotfunc(
                 darray = _rescale_imshow_rgb(darray, vmin, vmax, robust)
                 vmin, vmax, robust = None, None, False
 
+        if subplot_kws is None:
+            subplot_kws = dict()
+
+        if "surface" == plotfunc.__name__ and not kwargs.get("_is_facetgrid", False):
+            # Check we have new enough version of matplotlib
+            from distutils.version import LooseVersion
+
+            import matplotlib as mpl
+
+            if LooseVersion(mpl.__version__) < "3.2.0":
+                raise ValueError("surface plot requires at least matplotlib-3.2.0")
+            del LooseVersion
+            del mpl
```

Comment:
I think these were necessary, because of this a bit later
https://github.com/pydata/xarray/blob/648e13b0852fc89eb4d312a2b15d303f3f7a9ba7/xarray/plot/plot.py#L672-L673
If `LooseVersion` and `mpl` are not deleted, they are in the local environment, and end up in `allargs`, but are not valid keyword arguments.